pylint/checkers/typecheck.py

```diff
@@ -806,6 +806,10 @@ def _is_invalid_isinstance_type(arg: nodes.NodeNG) -> bool:
         return False
     if isinstance(inferred, astroid.Instance) and inferred.qname() == BUILTIN_TUPLE:
         return False
+    if isinstance(arg, nodes.BinOp) and arg.op == "|":
```

Comment:
I reckon this check should only be done for python version that support PEP604 and the `|` operand for type annotations.  I think this condition should only be checked when supported, similar code in `pylint/extensions/typing.py`

```suggestion
    if _should_check_alternative_union_syntax and isinstance(arg, nodes.BinOp) and arg.op == "|":
```

where:
```
        py_version = self.linter.config.py_version
        self._py37_plus = py_version >= (3, 7)
        self._py310_plus = py_version >= (3, 10)
        self._should_check_alternative_union_syntax = self._py310_plus or (
            self._py37_plus and self.linter.config.runtime_typing is False

```

Reply:
Agreed, I'll add that once the rest is figured out. And maybe a shorter name would be nice :laughing:

Reply:
maybe `_alt_union_syntax_enabled` ?

You can also rename it in `pylint/extensions/typing.py`